terraform/main.tf

```diff
@@ -18,3 +19,51 @@ output "serverless-secret-access-key" {
   value     = module.serverless-user.aws_secret_access_key
   sensitive = true
 }
+
+locals {
+  // app_name gets truncated to meet S3 bucket name limits, so the policy that the module creates doesn't match
+  s3_policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        Action = [
+          "s3:GetBucketLocation",
+          "s3:CreateBucket",
+          "s3:DeleteBucket",
```

Comment:
(I apologize if I should have caught this before.)
Does it need to be able to Delete Buckets?

Reply:
For `serverless remove` it does. If you think we should keep that out and require it be done manually, I'd be happy to consider that.

Reply:
It seems safer to leave it out.  @forevermatt , what do you think?

Reply:
OK. There are differences. I'll compare side-by-side and list out the differences.

Reply:
### Changed
- Changed service name from `serverless-vulnerability-scanner` to `vulnerability-scanner` because of S3 bucket name length restriction.
- Restricted `cloudformation:CreateUploadBucket` to `arn:aws:cloudformation:us-east-1:*:stack/vulnerability-scanner*/*` (was `*`).
- Changed to `cloudformation:Describe*` from specific describe actions.
- Changed to `lambda:Update*` from specific update actions.
- Removed resource restriction from `logsCreateLogGroup`
- Deployment bucket and results bucket now have separate policies. `s3:DeleteBucket` is on the deployment bucket but not on the results bucket.

### Added
- Resources now restricted by region
- Added `lambda:DeleteFunctionEventInvokeConfig`
- Added `iam:DetachRolePolicy`
- Added `logs:PutRetentionPolicy`
- Added `s3:PutBucketAcl`

### Removed
- Removed `events:Delete*`, `events:DescribeRule`, `events:Remove*`
- Removed `arn:aws:events:us-east-1::event-source/*`, `arn:aws:events:us-east-1:*:event-bus/*`
- Removed `lambda:UpdateEventSourceMapping`
- Removed `iam:GetRolePolicy`, `iam:ListRoleTags`, `iam:TagRole`, `iam:UntagRole`
- Removed `logs:CreateLogStream`, `logs:PutLogEvents`, `logs:FilterLogEvents`
- Removed `s3:DeleteBucketPolicy`, `s3:GetBucketPolicy`
- Removed `arn:aws:s3:::*` Resource from `s3:CreateBucket`

Reply:
Okay, thanks. Best I can tell, those are all fine.